mobius-core/src/test/java/com/spotify/mobius/MobiusLoopTest.java

```diff
@@ -440,20 +441,51 @@ public void eventsFromEffectHandlerDuringDisposeAreIgnored() throws Exception {
     assertFalse(updateWasCalled.get());
   }
 
+  @Test
+  public void modelsFromInitDuringDisposeAreIgnored() throws Exception {
+    // Model changes emitted from the init function during dispose should be ignored.
+
+    observer = new RecordingModelObserver<>();
+    Semaphore lock = new Semaphore(0);
+
+    final Update<String, TestEvent, TestEffect> update = (model, event) -> Next.noChange();
+    final MobiusLoop.Builder<String, TestEvent, TestEffect> builder =
+        Mobius.loop(update, effectHandler)
+            .init(
+                m -> {
+                  lock.acquireUninterruptibly();
+                  return First.first(m);
+                });
+
+    mobiusLoop = builder.startFrom("foo");
+    mobiusLoop.observe(observer);
+    releaseLockAfterDelay(lock, 30);
+    mobiusLoop.dispose();
+
+    observer.assertStates();
+  }
+
   @Test
   public void modelsFromUpdateDuringDisposeAreIgnored() throws Exception {
     // Model changes emitted from the update function during dispose should be ignored.
 
     observer = new RecordingModelObserver<>();
     Semaphore lock = new Semaphore(0);
 
+    final Update<String, TestEvent, TestEffect> update =
+        (model, event) -> {
+          lock.acquireUninterruptibly();
+          return Next.next("baz");
+        };
+
     final MobiusLoop.Builder<String, TestEvent, TestEffect> builder =
-        Mobius.loop(
-            (model, event) -> {
-              lock.acquireUninterruptibly();
-              return Next.next("baz");
-            },
-            effectHandler);
+        Mobius.loop(update, effectHandler)
+            .eventRunner(
+                () -> {
+                  final WorkRunner eventRunner =
+                      WorkRunners.from(Executors.newSingleThreadExecutor());
+                  return InitImmediatelyThenUpdateConcurrentlyWorkRunner.create(eventRunner);
```

Comment:
Could use the `backgroundRunner` here instead of creating a new one?

Reply:
hmm I noticed this is a member of the class, and isn't recreated for every test in setup. Wouldn't that lead to possible problems across test runs?